applications/dashboard/controllers/class.usercontroller.php

```diff
@@ -1071,14 +1072,18 @@ public function sso($UserID = false) {
     }
 
     /**
+     * JSON output of a username search.
      *
-     *
-     * @param $q
+     * @param string $query
      * @param int $limit
      */
-    public function tagSearch($q, $limit = 10) {
-        $Data = Gdn::userModel()->tagSearch($q, $limit);
-        die(json_encode($Data));
+    public function tagSearch($query, $limit = 10) {
+        $data = Gdn::userModel()->tagSearch($query, $limit);
+        foreach ($data as &$user) {
+            // Sanitize usernames.
```

Comment:
header('Content-type: application/json');  should be the right way to do this. Htmlescape is not needed